ts/components/messages/MessageCTABar.tsx

```diff
@@ -141,13 +141,55 @@ class MessageCTABar extends React.PureComponent<Props, State> {
     return this.props.payment !== undefined;
   }
 
+  private searchEventInCalendar = async (
```

Comment:
Could you provide a comment describing what this function does?
+
Since we prefer a more functional approach, what do you think about this implementation (same logic)?

```typescript
private searchEventInCalendar = async (
    endDate: Date,
    title: string
  ): Promise<Option<string>> => {
    const startDate = new Date(endDate.getTime());
    return RNCalendarEvents.fetchAllEvents(
      formatDateAsReminder(new Date(startDate.setDate(endDate.getDate() - 1))),
      formatDateAsReminder(endDate)
    )
      .then(
        events => {
          return fromNullable(events)
            .mapNullable(evs =>
              evs.find(e => {
                return (
                  e.title === title &&
                  new Date(e.endDate).getDay() === endDate.getDay()
                );
              })
            )
            .map(ev => some(ev.id))
            .getOrElse(none);
        },
        // handle promise rejection
        () => {
          return none;
        }
      )
      .catch(() => none);
  };
```